workout/templates/workout/workout_details.html

```diff
@@ -65,100 +58,120 @@ <h1>Description: <span>{{ workout.description }}</span></h1>
+                    hidelabel: 'block',
+                    modals: 'none',
+                }
+            },
+            watch: {
+                currentprofile:{
+                    handler(newValue, oldValue) {
+                        if (this.currentprofile.usertype == 1) {
+                            this.currentprofile.user_type = 'Trainee'
+                        } else {
+                            this.currentprofile.user_type = 'Trainer'
+                        }
+                        if(this.currentprofile.gender == 'M'){
+                            this.currentprofile.sex = 'Male'
+                        } else{
+                            this.currentprofile.sex = 'Female'
+                        }
+                    },
+                    deep: true
+                } ,
+            },
+            mounted: function(){
+                this.getWorkouts();
+                this.getCurrentWorkout(workoutid);
+                this.getCurrentProfile(profileId);
+                this.getCurrentUser(userId);
+            },
+            methods: {
+                getCurrentWorkout: function(id){
+                    var resource = this.$resource('/api/workout{/id}');
+                    resource.get({id:id})
+                        .then((response) => {
+                            this.currentworkout = response.data;
+                        }).catch((error) => {
+                            console.log('ERROR')
+                        });
+                },
+                getWorkouts: function(e){
+                    var resource = this.$resource('/api/workout/');
+                    resource.get()
+                        .then((response)=> {
+                            this.workouts = response.data;
+                        }).catch((error) => {
+                            console.log('ERROR')
+                        });
+                },
+                user: function(){
+                    this.modals = 'block'
+                },
+                close: function(){
+                    this.modals = 'none'
+                },
+                showForms: function(){
+                    this.showform = 'block'
+                    this.hidelabel = 'none'
+                    this.getCurrentProfile(1)
+                    return
+                },
+                showLabels: function(){
+                    this.showform = 'none'
+                    this.hidelabel = 'block'
+                },
+                updateProfile: function(id){
+                    getCookie;
+                    this.$http.put('/api/userprofile{/id}/update/', this.currentprofile, {headers: {'X-CSRFToken': getCookie("csrftoken")}})
+                    .then((response) => {
+                        this.currentprofile = {};
+                    }).catch((error) => {
+                            console.log('ERROR')
+                        });
+                },
+                getCurrentProfile: function(user) {
```

Comment:
@regilkent123 update this to make use of the response to also set the current user